### PR TITLE
Update to 3.25.1...

### PR DESCRIPTION
**Pull request by ocefpaf**
... And add `SQLITE_ENABLE_GEOPOLY` support.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
+export CPPFLAGS="${CPPFLAGS} -DSQLITE_ENABLE_COLUMN_METADATA=1 \
+                             -DSQLITE_ENABLE_UNLOCK_NOTIFY \
+                             -DSQLITE_ENABLE_DBSTAT_VTAB=1 \
+                             -DSQLITE_ENABLE_FTS3_TOKENIZER=1 \
+                             -DSQLITE_SECURE_DELETE \
+                             -DSQLITE_MAX_VARIABLE_NUMBER=250000 \
+                             -DSQLITE_MAX_EXPR_DEPTH=10000 \
+                             -DSQLITE_ENABLE_GEOPOLY \
+                             -DSQLITE_ENABLE_JSON1 \
+                             -DSQLITE_ENABLE_RTREE=1"
```
Comment (ocefpaf) [inline recipe/build.sh]: @snorfalorpagus do you know if there are any simple tests we can do to test RTREE abd GEOPOLY support here?

File: recipe/build.sh [hunk 2]
```diff
@@ -1,24 +1,42 @@
...
-            $B --enable-threadsafe \
-            --enable-json1 \
-            --enable-tempstore \
+./configure --prefix=${PREFIX} \
+            --build=${BUILD} \
+            --host=${HOST} \
+            --enable-threadsafe \
             --enable-shared=yes \
             --enable-readline \
             --disable-editline \
```
Comment (ocefpaf) [inline recipe/build.sh]: @msarahan and @mingwandroid I noticed that in AR version of this recipe you have:

```
            --enable-readline    \
            --enable-editline    \
            --disable-readline   \
```

and `libedit` is used in the recipe.

I wonder what is the reason and if we should follow it here or not.
Ping @conda-forge/core

Comment (ocefpaf) [reply recipe/build.sh]: That part I know :smile: 

I wonder if we should do it in `conda-forge` too. I hear bad things about `libedit` but I never really used it. If that is "good enough" I guess we could switch.

**Comment by mingwandroid** [comment]
Libedit or readline are only used by the cli programs, so I don't think it's very important which is used here.

@jjhelmus had some preference (is it weren't for the licence issue)!for readline but I don't know the details.

File: recipe/build.sh (hunk 2 above)
Comment (jjhelmus) [reply recipe/build.sh]: Personally, I don't use the `sqlite` REPL so I have no need for `readline` or `libedit` support.  Hearing the opinions of folks who do make use of the REPL would be helpful in this decision.  

For Python, I am in favor of keeping `readline`. The Python REPL with libedit is less functional that when using `libedit`.

**Comment by ocefpaf** [comment]
> Libedit or readline are only used by the cli programs, so I don't think it's very important which is used here.
@jjhelmus had some preference (is it weren't for the licence issue)!for readline but I don't know the details.

Thanks for the info! I'll merge this one with `readline` and test `libedit` for `sqlite` in the future. Maybe the approach in #32 would make everybody happy when CLI is important.

**Comment by mingwandroid** [comment]
Beware that using readline in anything makes that thing GPL.

**Comment by mingwandroid** [comment]
So you can no longer state that sqlite is BSD.

**Comment by mingwandroid** [comment]
.. or anything it then links to.

File: recipe/build.sh (hunk 2 above)
Comment (ocefpaf) [reply recipe/build.sh]: @mingwandroid I guess that the first `--enable-readline` there is unnecessary, right? All we need to try `libedit` would be:

```
--enable-editline    \
--disable-readline   \
```

correct?

Comment (ocefpaf) [reply recipe/build.sh]: > For Python, I am in favor of keeping `readline`. The Python REPL with `libedit` is less functional that when using `libedit`.

That is what concerns me when considering moving to `libedit`. You are not the only one I know that complains about it and prefers `readline`. So let's not touch that for now :grimacing:

Regarding the licenses, if one interprets the metadata in the recipe are the Software license, then we are good, but if that is interpreted as the final **package** license, then we probably say GPL instead. But that is a discussion for later :smile:

**Comment by mingwandroid** [comment]
I feel on AD at least we have no choice but to remove all GPL dependencies from non-GPL software.

LGPL doesn't require this fortunately but readline is not LGPL.

**Comment by ocefpaf** [comment]
> I feel on AD at least we have no choice but to remove all GPL dependencies from non-GPL software.

Sorry. Not sure what AD means there :smile: (Remember I'm an old guy and I don't know these cool acronyms young people like you use :stuck_out_tongue_winking_eye:)

Anyway, I believe we need a long term plan to address things like that.
I'll bring that up in our next meeting.

**Comment by ocefpaf** [comment]
> AD=Anaconda Distribution, trying to save my fingers.

Sorry to make you write more :grimacing:

**Pull request merged**